File: executable/visual-debugger/src/main/java/org/visual/debugger/factory/JvmFactory.java
package org.visual.debugger.factory;

import io.avaje.inject.Bean;
import io.avaje.inject.Factory;
import io.github.classgraph.ClassGraph;
import java.lang.management.*;
import lombok.extern.slf4j.Slf4j;

@Factory
@Slf4j
public class JvmFactory {

  @Bean
  MemoryMXBean memoryMXBean() {
    return ManagementFactory.getMemoryMXBean();
  }

  @Bean
  ClassLoadingMXBean classLoadingMXBean() {
    return ManagementFactory.getClassLoadingMXBean();
  }

  @Bean
  CompilationMXBean compilationMXBean() {
    return ManagementFactory.getCompilationMXBean();
  }

  @Bean
  RuntimeMXBean runtimeMXBean() {
    return ManagementFactory.getRuntimeMXBean();
  }

  @Bean
  OperatingSystemMXBean operatingSystemMXBean() {
    return ManagementFactory.getOperatingSystemMXBean();
  }

  @Bean
  ThreadMXBean threadMXBean() {
    return ManagementFactory.getThreadMXBean();
  }

  @Bean
  PlatformLoggingMXBean platformLoggingMXBean() {
    return ManagementFactory.getPlatformMXBean(PlatformLoggingMXBean.class);
  }

  @Bean
  ClassGraph classGraph() {
    return new ClassGraph().enableAllInfo();
  }
}

File: ui/visual-debugger/src/main/kotlin/org/visual/debugger/module/JvmFactory.kt
package org.visual.debugger.module

import com.google.inject.AbstractModule
import com.google.inject.Provides
import io.github.classgraph.ClassGraph
import java.lang.management.*

class JvmFactory :AbstractModule(){

    @Provides
    fun memoryMXBean(): MemoryMXBean {
        return ManagementFactory.getMemoryMXBean()
    }

    @Provides
    fun classLoadingMXBean(): ClassLoadingMXBean {
        return ManagementFactory.getClassLoadingMXBean()
    }

    @Provides
    fun compilationMXBean(): CompilationMXBean {
        return ManagementFactory.getCompilationMXBean()
    }

    @Provides
    fun runtimeMXBean(): RuntimeMXBean {
        return ManagementFactory.getRuntimeMXBean()
    }

    @Provides
    fun operatingSystemMXBean(): OperatingSystemMXBean {
        return ManagementFactory.getOperatingSystemMXBean()
    }

    @Provides
    fun threadMXBean(): ThreadMXBean {
        return ManagementFactory.getThreadMXBean()
    }

    @Provides
    fun platformLoggingMXBean(): PlatformLoggingMXBean {
        return ManagementFactory.getPlatformMXBean(PlatformLoggingMXBean::class.java)
    }

    @Provides
    fun classGraph(): ClassGraph {
        return ClassGraph().enableAllInfo()
    }
}
